app/monitoringapi_internal_test.go

```diff
@@ -91,8 +97,8 @@ func TestStartChecker(t *testing.T) {
 			}
 
 			// connect each host with its peers
-			for i := 0; i < tt.numPeers; i++ {
-				for k := tt.absentPeers; k < tt.numPeers; k++ {
+			for i := 0; i < tt.numPeers-tt.absentPeers; i++ {
```

Comment:
note I swapped this since we now open connections, not just add addresses, and adding a connection results in both sides of the connection being "active". so you are effectively connecting peer 0 to all peers even if absentPeers > 0

Reply:
so if number of peers are 4 and absent peers is 1 then according this:
0 is connected to 1 and 2
1 is connected to 0 and 2 
2 is connected to 0 and 1
3 is connected to none

peer 0 is not connected to peer 3

Reply:
I have added tt.numPeers-tt.absentPeers condition to both the for loops "since we now open connections".

Reply:
According to your logic, if we have 4 peers and absent peers is 1 then:
0 is connected to 1, 2 and 3
1 is connected to 2 and 3
2 is connected to 1 and 3
3 is connected to 1 and 2

we call `startReadyChecker` for host[0] for which count in `quorumPeersConnected` function comes out to be 3. However, 1 peer should be absent and count should be 2.

Reply:
Maybe to simplify, since we only test node0, is to only connect node0 to N peers. Do not also connect the other needs to each other.

Reply:
sure

Reply:
done